vmas/simulator/environment/environment.py

```diff
@@ -740,6 +740,51 @@ def render(
                 pos[Y] + cam_range[Y],
             )
 
+        # Include boundaries in the rendering
+        from vmas.simulator import rendering
+        from vmas.simulator.rendering import Line
+        from vmas.simulator.utils import Color
+
+        # Check if the world dimensions are defined
+        if self.world.x_semidim is not None and self.world.y_semidim is not None:
+            # Get the radius of the agents
+            radius = self.agents[0].shape.radius
```

Comment:
Which scenario are you referring to? Just to have a concrete example, thanks.

This is purely a visualization issue, not something more, since I believe that (center + half_radius) should never exceed the boundary limits. Right?

The safest approach would be to use the max_radius (where can I get it?), but this could cause some visualization issues, such as smaller agents not reaching the visualized boundary.

Suggestions? :)

Reply:
exactly,

with 2 you have smaller agents not reaching the boundary, with 1 the boundary is always for the center of agents

this is partly what concerns me about this PR: it is impossible to have a boundary that fits all agents